src/main/java/net/laboulangerie/laboulangeriemmo/commands/TownyMmo.java

```diff
@@ -1,214 +1,158 @@
+            if (nationTopCache.get(args[1]) == null) {
+                if(args[1].equals("total")) {
+                	nationTopCache.put(args[1], List.of(TownyUniverse.getInstance().getNations()).stream().map(nation ->
+                	TownyUniverse.getInstance().getNation(nation.toArray()[0].toString())
+            			).sorted((v1, v2) ->  (MmoPlayer.getNationTotalLevel(v2)).compareTo(MmoPlayer.getNationTotalLevel(v1))).collect(Collectors.toList()));
                 }
-                if (args[2].equalsIgnoreCase("mining") || args[2].equalsIgnoreCase("farmer") || args[2].equalsIgnoreCase("woodcutting") || args[2].equalsIgnoreCase("thehunter")) {
-                    int total = 0;
-                    Nation nation = TownyUniverse.getInstance().getNation(args[3]);
-                    String nationUnName = nation.getName();
-                    total = MmoPlayer.getNationTalentLevel(nation, args[2]);
-                    sender.sendMessage("Le niveau total de " + nationUnName + "dans le métier " + args[2] + " est de : " + total);
+                else {
+                	nationTopCache.put(args[1], List.of(TownyUniverse.getInstance().getNations()).stream().map(nation ->
+                	TownyUniverse.getInstance().getNation(nation.toArray()[0].toString())
+            			).sorted((v1, v2) ->  (MmoPlayer.getNationTalentLevel(v2, args[1])).compareTo(MmoPlayer.getNationTalentLevel(v1, args[1]))).collect(Collectors.toList()));
                 }
+                scheduleCacheClear(args[1]);
+            }
+            List<Nation> orderedNations = nationTopCache.get(args[1]);
+
+            if(args[1].equals("total")) {
+            	sender.sendMessage("§3----------§8[Page §7" + (page+1) + "§8]§3----------");
+            	for (int i = page*10; i < (orderedNations.size() < (page+1) *10 ? orderedNations.size() : (page+1) *10); i++) {
+                	Nation nation = orderedNations.get(i);
+                	sender.sendMessage("§e" + (i+1) + ". §a" + nation.getName() + " §6- §3total §9" + MmoPlayer.getNationTotalLevel(nation));
+            	}
+            }
+            else {
+            	sender.sendMessage("§3----------§8[Page §7" + (page+1) + "§8]§3----------");
+            	for (int i = page*10; i < (orderedNations.size() < (page+1) *10 ? orderedNations.size() : (page+1) *10); i++) {
+                	Nation nation = orderedNations.get(i);
+                	sender.sendMessage("§e" + (i+1) + ". §a" + nation.getName() + " §6- §3level §9" + MmoPlayer.getNationTalentLevel(nation, args[1]));
+            	}
             }
         }
-        return false;
+
+        return true;
+    }
+    private void scheduleCacheClear(String talent) {
+        new BukkitRunnable() {
+            @Override
+            public void run() {
+                townTopCache.remove(talent);
+                nationTopCache.remove(talent);
+            }
+        }.runTaskLater(LaBoulangerieMmo.PLUGIN, 20*60);
```

Comment:
```suggestion
        Bukst.getScheduler().runTaskLater(LaBoulangerieMmo.PLUGIN, () -> {
                townTopCache.remove(talent);
                nationTopCache.remove(talent);
        }, 20*60);
```

Instead of creating an anonymous class better parse runnable as function into the Scheduler